purchases/src/main/kotlin/com/revenuecat/purchases/common/networking/RCHTTPStatusCodes.kt

```diff
@@ -11,4 +11,9 @@ internal object RCHTTPStatusCodes {
 
     fun isSuccessful(statusCode: Int) = statusCode < BAD_REQUEST
     fun isServerError(statusCode: Int) = statusCode >= ERROR
+
+    // Note: this means that all 4xx (except 404) are considered as successfully synced.
+    // The reason is because it's likely due to a client error, so continuing to retry
+    // won't yield any different results and instead kill pandas.
+    fun isSynced(statusCode: Int) = isSuccessful(statusCode) || !(isServerError(statusCode) || statusCode == NOT_FOUND)
```

Comment:
I'm surprised this didn't exist yet. We use it in iOS for marking attributes as synced.

Reply:
Hmm that's slightly different in Android for subscriber attributes... There, we would mark them as synced if it's not a server error and we receive a backend error code 24 (UnsupportedError), but we don't mark them as synced with 404... I guess we could add that as well?